dev/RadioButtons/APITests/RadioButtonsTests.cs

```diff
@@ -0,0 +1,83 @@
+
+using MUXControlsTestApp.Utilities;
+
+using Windows.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls;
+using Common;
+using Windows.UI.Xaml.Markup;
+using System.Collections.Generic;
+using Windows.UI.Xaml.Media;
+
+#if USING_TAEF
+using WEX.TestExecution;
+using WEX.TestExecution.Markup;
+using WEX.Logging.Interop;
+#else
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.TestTools.UnitTesting.Logging;
+#endif
+
+namespace Windows.UI.Xaml.Tests.MUXControls.ApiTests
+{
+    [TestClass]
+    public class RadioButtonsTests : ApiTestBase
+    {
+        [TestMethod]
+        public void VerifyCustomItemTemplate()
+        {
+            RadioButtons radioButtons = null;
+            RadioButtons radioButtons2 = null;
+            RunOnUIThread.Execute(() =>
+            {
+                radioButtons = new RadioButtons();
+                radioButtons.ItemsSource = new List<string>() { "Option 1", "Option 2" };
+
+                // Set a custom ItemTemplate to be wrapped in a RadioButton.
+                var itemTemplate = (DataTemplate)XamlReader.Load(
+                        @"<DataTemplate xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'>
+                            <TextBlock Text = '{Binding}'/>
+                        </DataTemplate>");
+
+                radioButtons.ItemTemplate = itemTemplate;
+
+                radioButtons2 = new RadioButtons();
+                radioButtons2.ItemsSource = new List<string>() { "Option 1", "Option 2" };
+
+                // Set a custom ItemTemplate which is already a RadioButton. No wrapping should be performed.
+                var itemTemplate2 = (DataTemplate)XamlReader.Load(
+                        @"<DataTemplate xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'>
+                            <RadioButton Foreground='Blue'>
+                              <TextBlock Text = '{Binding}'/>
+                            </RadioButton>
+                        </DataTemplate>");
+
+                radioButtons2.ItemTemplate = itemTemplate2;
+
+                var stackPanel = new StackPanel();
+                stackPanel.Children.Add(radioButtons);
+                stackPanel.Children.Add(radioButtons2);
+
+                Content = stackPanel;
+                Content.UpdateLayout();
+            });
+
+            IdleSynchronizer.Wait();
+
+            RunOnUIThread.Execute(() =>
+            {
+                var radioButton1 = radioButtons.ContainerFromIndex(0) as RadioButton;
+                var radioButton2 = radioButtons2.ContainerFromIndex(0) as RadioButton;
+                Verify.IsNotNull(radioButton1, "Our custom ItemTemplate should have been wrapped in a RadioButton.");
+                Verify.IsNotNull(radioButton2, "Our custom ItemTemplate should have been wrapped in a RadioButton.");
+
+                bool testCondition = !(radioButton1.Foreground is SolidColorBrush brush && brush.Color == Colors.Blue);
```

Comment:
>testCondition [](start = 21, length = 13)

Why do you prefer this pattern? (not saying its bad, just curious)

Reply:
First of all, Brush.Equals() doesn't appear to work here so I can't use that:
```cs
var brush1 = new SolidColorBrush(Colors.Blue);
var brush2 = new SolidColorBrush(Colors.Blue);

bool b = brush1.Equals(brush2); // b = false
```

Secondly, I think it is easier to read if the condition we want to test has its own statement to keep it separate from the actual verification *and* whatever the consequence is if verification fails (mostly logging a message passed to the Verify() call). That way, the focus of the reader can be solely guided to the testCondition here, which is the most significant part of a Verify() call (*what* do we want to verify here)?

Lastly, instead of using both `Verify.IsFalse()` *and* `Verify.IsTrue()` here I want to stick with `Verify.IsTrue()` to guide the focus of the reader solely to the "testCondition" and it's value. I don't want a Verify.IsTrue(testCondition) to be followed by a `testCondition = ....; Verify.IsFalse(testCondition)` as then not only would there be a different *testCondition*, but also one case where we test if it's *true* and another case where we test if it is *false*. If I keep both Verify calls consistent (so Verify.IsTrue()) here, the reader only needs to pay attention to the updated *testCondition*.

Reply:
brush1.Equals(brush2) is doing an object comparison, so it makes sense that this returns false, they aren't the same object, they just have the same color value.

You could do something like 
```
var radioButtonForegroundBrush = RadioButton1.Foreground as SolidColorBrush;
Verify.isNotNull(radioButtonForegroundBrush);
Verify.isTrue(radioButtonForegroundBrush.Color == Colors.Blue);
```

The benefit of this is that the test output will tell you which of the verifies failed, where as with your version if the verify fails you won't be able to tell from the output if it was caused by the brush not being a SolidColorBrush or the color not being blue.

Reply:
@StephenLPeters 

> brush1.Equals(brush2) is doing an object comparison, so it makes sense that this returns false, they aren't the same object, they just have the same color value.

I would have expected something like [this](https://docs.microsoft.com/en-us/dotnet/api/system.object.equals?view=netcore-3.1#System_Object_Equals_System_Object_) here:
```cs
public override bool Equals(Object obj)
   {
      //Check for null and compare run-time types.
      if ((obj == null) || ! this.GetType().Equals(obj.GetType()))
      {
         return false;
      }
      else {
         Point p = (Point) obj;
         return (x == p.x) && (y == p.y);
      }
   }
```

(Since the properties are the same, the brushes are equal to me even though they are different objects.)

> You could do something like

Is that necessary here? I more or less only care about whether the updated RadioButtons.Foreground now matches my test brush or not. If it doesn't, then that should already be enough info for me here.